packages/create-redwood-app/src/create-redwood-app.js

```diff
@@ -67,6 +67,33 @@ export const CreateNewApp = ({ args }) => {
     setMessages(latestMessages.current)
   }
 
+  // set index.html title based on user input directory
+  const changeHtmlTitle = async (path) => {
+    const newTitle = (userDir) => {
+      return userDir
+        .split("/")
+        .slice(-1)[0]
+        .split(/ |-|_/)
+        .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
+        .join(' ')
+    }
+
+    fs.readFile(path + '/web/src/index.html', 'utf8', (e, data) => {
+      if (e) {
+        return setNewMessage(e)
```

Comment:
`setNewMessage()` doesn't have any return value, what's the intent here?

Reply:
I was originally using `console.log(e)` to return error messages. But it seemed they were getting lost in the async madness, if displaying at all when errors occurred. 

Is there a better way to handle errors than `setNewMessage(e)` here?

Reply:
@peterp can you shed some light on this?

Reply:
Each time `setNewMessage` is called the screen is redrawn, so a `console.log` would output to the terminal and then `setNewMessage` would overwrite it. 

I think @thedavidprice is setting the message here, and returning early, so that it's displaying the error output:
```js
setNewMessage(e.message)
return // abort early.
```

Using `fs.readFileSync` / `fs.writeFileSync` would also make this easier to deal with, and you could remove the `await`/ `async` stuff since `changeHtmlTitle(newAppDir)` is not returning a promise.

But, the root cause  is that the code I wrote is a nightmare, the implementation of `setNewMessage` is what Dan describes as "[swimming against the tide](https://overreacted.io/a-complete-guide-to-useeffect/#swimming-against-the-tide)", and we should refactor this code to be easier to reason about, I would replace this implementation to use [Listr](https://github.com/SamVerschueren/listr) instead of React Ink.

Reply:
Hey now, if _anyone_ gets to talk trash about their own code, it's me! 😉

Definitely don't put this on your list. I have several ways forward here — let me evaluate my time in a few days and choose an appropriate approach. 
- [ ] given time, look into refactoring using Listr
- [ ] promises look, ahem, promising given PP code below
- [ ] read/writeFileSync likely the fastest way forward.

PS "A Complete Guide to useEffect" is great! Thanks for referencing.